Model Class generating with Error

### Before opening, please confirm:


- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-codegen/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#bugs).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

_No response_

### If applicable, what version of Node.js are you using?

_No response_

### Amplify CLI Version

8.0.0

### What operating system are you using?

Linux Mint

### Amplify Codegen Command

codegen models

### Describe the bug

### Before opening, please confirm:


- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Java

### Amplify Categories

Storage

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>


### Environment information

<details>

```
# Put output below this line


```

</details>


### Please include any relevant guides or documentation you're referencing

_No response_

### Describe the bug

I have created one schema with the name of GoalStatus. But when I run the command **amplify codegen models**
Model class generated but it gives an error.

```
/app/src/main/java/com/amplifyframework/datastore/generated/model/GoalStatus.java:408: error: cannot find symbol
        .startDate(startDate)
        ^
  symbol:   method startDate(DateTime)
  location: interface TaskReferencesStep


```
![Screenshot from 2022-04-13 11-44-10](https://user-images.githubusercontent.com/13220264/163112389-0507eb78-1544-4cc1-abb0-83bc216c3941.png)



```
type GoalStatus @model @auth(rules: [{ allow: owner }]) {
id: ID!
username: String!
version: Int!
goalReference: String!
goalText: String
taskReferences: [String]!
startDate: AWSDateTime!
dueDate: AWSDateTime!
recordedDate: AWSDateTime
complete: Boolean!
selected: Boolean!
goalAdditionalInfo:String
}
```
```
package com.amplifyframework.datastore.generated.model;

import com.amplifyframework.core.model.temporal.Temporal;

import java.util.List;
import java.util.UUID;
import java.util.Objects;

import androidx.core.util.ObjectsCompat;

import com.amplifyframework.core.model.AuthStrategy;
import com.amplifyframework.core.model.Model;
import com.amplifyframework.core.model.ModelOperation;
import com.amplifyframework.core.model.annotations.AuthRule;
import com.amplifyframework.core.model.annotations.Index;
import com.amplifyframework.core.model.annotations.ModelConfig;
import com.amplifyframework.core.model.annotations.ModelField;
import com.amplifyframework.core.model.query.predicate.QueryField;

import static com.amplifyframework.core.model.query.predicate.QueryField.field;

/** This is an auto generated class representing the GoalStatus type in your schema. */
@SuppressWarnings("all")
@ModelConfig(pluralName = "GoalStatuses", authRules = {
  @AuthRule(allow = AuthStrategy.OWNER, ownerField = "owner", identityClaim = "cognito:username", provider = "userPools", operations = { ModelOperation.CREATE, ModelOperation.UPDATE, ModelOperation.DELETE, ModelOperation.READ })
})
public final class GoalStatus implements Model {
  public static final QueryField ID = field("GoalStatus", "id");
  public static final QueryField USERNAME = field("GoalStatus", "username");
  public static final QueryField VERSION = field("GoalStatus", "version");
  public static final QueryField GOAL_REFERENCE = field("GoalStatus", "goalReference");
  public static final QueryField GOAL_TEXT = field("GoalStatus", "goalText");
  public static final QueryField TASK_REFERENCES = field("GoalStatus", "taskReferences");
  public static final QueryField START_DATE = field("GoalStatus", "startDate");
  public static final QueryField DUE_DATE = field("GoalStatus", "dueDate");
  public static final QueryField RECORDED_DATE = field("GoalStatus", "recordedDate");
  public static final QueryField COMPLETE = field("GoalStatus", "complete");
  public static final QueryField SELECTED = field("GoalStatus", "selected");
  public static final QueryField GOAL_ADDITIONAL_INFO = field("GoalStatus", "goalAdditionalInfo");
  private final @ModelField(targetType="ID", isRequired = true) String id;
  private final @ModelField(targetType="String", isRequired = true) String username;
  private final @ModelField(targetType="Int", isRequired = true) Integer version;
  private final @ModelField(targetType="String", isRequired = true) String goalReference;
  private final @ModelField(targetType="String") String goalText;
  private final @ModelField(targetType="String", isRequired = true) List<String> taskReferences;
  private final @ModelField(targetType="AWSDateTime", isRequired = true) Temporal.DateTime startDate;
  private final @ModelField(targetType="AWSDateTime", isRequired = true) Temporal.DateTime dueDate;
  private final @ModelField(targetType="AWSDateTime") Temporal.DateTime recordedDate;
  private final @ModelField(targetType="Boolean", isRequired = true) Boolean complete;
  private final @ModelField(targetType="Boolean", isRequired = true) Boolean selected;
  private final @ModelField(targetType="String") String goalAdditionalInfo;
  private @ModelField(targetType="AWSDateTime", isReadOnly = true) Temporal.DateTime createdAt;
  private @ModelField(targetType="AWSDateTime", isReadOnly = true) Temporal.DateTime updatedAt;
  public String getId() {
      return id;
  }
  
  public String getUsername() {
      return username;
  }
  
  public Integer getVersion() {
      return version;
  }
  
  public String getGoalReference() {
      return goalReference;
  }
  
  public String getGoalText() {
      return goalText;
  }
  
  public List<String> getTaskReferences() {
      return taskReferences;
  }
  
  public Temporal.DateTime getStartDate() {
      return startDate;
  }
  
  public Temporal.DateTime getDueDate() {
      return dueDate;
  }
  
  public Temporal.DateTime getRecordedDate() {
      return recordedDate;
  }
  
  public Boolean getComplete() {
      return complete;
  }
  
  public Boolean getSelected() {
      return selected;
  }
  
  public String getGoalAdditionalInfo() {
      return goalAdditionalInfo;
  }
  
  public Temporal.DateTime getCreatedAt() {
      return createdAt;
  }
  
  public Temporal.DateTime getUpdatedAt() {
      return updatedAt;
  }
  
  private GoalStatus(String id, String username, Integer version, String goalReference, String goalText, List<String> taskReferences, Temporal.DateTime startDate, Temporal.DateTime dueDate, Temporal.DateTime recordedDate, Boolean complete, Boolean selected, String goalAdditionalInfo) {
    this.id = id;
    this.username = username;
    this.version = version;
    this.goalReference = goalReference;
    this.goalText = goalText;
    this.taskReferences = taskReferences;
    this.startDate = startDate;
    this.dueDate = dueDate;
    this.recordedDate = recordedDate;
    this.complete = complete;
    this.selected = selected;
    this.goalAdditionalInfo = goalAdditionalInfo;
  }
  
  @Override
   public boolean equals(Object obj) {
      if (this == obj) {
        return true;
      } else if(obj == null || getClass() != obj.getClass()) {
        return false;
      } else {
      GoalStatus goalStatus = (GoalStatus) obj;
      return ObjectsCompat.equals(getId(), goalStatus.getId()) &&
              ObjectsCompat.equals(getUsername(), goalStatus.getUsername()) &&
              ObjectsCompat.equals(getVersion(), goalStatus.getVersion()) &&
              ObjectsCompat.equals(getGoalReference(), goalStatus.getGoalReference()) &&
              ObjectsCompat.equals(getGoalText(), goalStatus.getGoalText()) &&
              ObjectsCompat.equals(getTaskReferences(), goalStatus.getTaskReferences()) &&
              ObjectsCompat.equals(getStartDate(), goalStatus.getStartDate()) &&
              ObjectsCompat.equals(getDueDate(), goalStatus.getDueDate()) &&
              ObjectsCompat.equals(getRecordedDate(), goalStatus.getRecordedDate()) &&
              ObjectsCompat.equals(getComplete(), goalStatus.getComplete()) &&
              ObjectsCompat.equals(getSelected(), goalStatus.getSelected()) &&
              ObjectsCompat.equals(getGoalAdditionalInfo(), goalStatus.getGoalAdditionalInfo()) &&
              ObjectsCompat.equals(getCreatedAt(), goalStatus.getCreatedAt()) &&
              ObjectsCompat.equals(getUpdatedAt(), goalStatus.getUpdatedAt());
      }
  }
  
  @Override
   public int hashCode() {
    return new StringBuilder()
      .append(getId())
      .append(getUsername())
      .append(getVersion())
      .append(getGoalReference())
      .append(getGoalText())
      .append(getTaskReferences())
      .append(getStartDate())
      .append(getDueDate())
      .append(getRecordedDate())
      .append(getComplete())
      .append(getSelected())
      .append(getGoalAdditionalInfo())
      .append(getCreatedAt())
      .append(getUpdatedAt())
      .toString()
      .hashCode();
  }
  
  @Override
   public String toString() {
    return new StringBuilder()
      .append("GoalStatus {")
      .append("id=" + String.valueOf(getId()) + ", ")
      .append("username=" + String.valueOf(getUsername()) + ", ")
      .append("version=" + String.valueOf(getVersion()) + ", ")
      .append("goalReference=" + String.valueOf(getGoalReference()) + ", ")
      .append("goalText=" + String.valueOf(getGoalText()) + ", ")
      .append("taskReferences=" + String.valueOf(getTaskReferences()) + ", ")
      .append("startDate=" + String.valueOf(getStartDate()) + ", ")
      .append("dueDate=" + String.valueOf(getDueDate()) + ", ")
      .append("recordedDate=" + String.valueOf(getRecordedDate()) + ", ")
      .append("complete=" + String.valueOf(getComplete()) + ", ")
      .append("selected=" + String.valueOf(getSelected()) + ", ")
      .append("goalAdditionalInfo=" + String.valueOf(getGoalAdditionalInfo()) + ", ")
      .append("createdAt=" + String.valueOf(getCreatedAt()) + ", ")
      .append("updatedAt=" + String.valueOf(getUpdatedAt()))
      .append("}")
      .toString();
  }
  
  public static UsernameStep builder() {
      return new Builder();
  }
  
  /** 
   * WARNING: This method should not be used to build an instance of this object for a CREATE mutation.
   * This is a convenience method to return an instance of the object with only its ID populated
   * to be used in the context of a parameter in a delete mutation or referencing a foreign key
   * in a relationship.
   * @param id the id of the existing item this instance will represent
   * @return an instance of this model with only ID populated
   */
  public static GoalStatus justId(String id) {
    return new GoalStatus(
      id,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null
    );
  }
  
  public CopyOfBuilder copyOfBuilder() {
    return new CopyOfBuilder(id,
      username,
      version,
      goalReference,
      goalText,
      taskReferences,
      startDate,
      dueDate,
      recordedDate,
      complete,
      selected,
      goalAdditionalInfo);
  }
  public interface UsernameStep {
    VersionStep username(String username);
  }
  

  public interface VersionStep {
    GoalReferenceStep version(Integer version);
  }
  

  public interface GoalReferenceStep {
    TaskReferencesStep goalReference(String goalReference);
  }
  

  public interface TaskReferencesStep {
    StartDateStep taskReferences(List<String> taskReferences);
  }
  

  public interface StartDateStep {
    DueDateStep startDate(Temporal.DateTime startDate);
  }
  

  public interface DueDateStep {
    CompleteStep dueDate(Temporal.DateTime dueDate);
  }
  

  public interface CompleteStep {
    SelectedStep complete(Boolean complete);
  }
  

  public interface SelectedStep {
    BuildStep selected(Boolean selected);
  }
  

  public interface BuildStep {
    GoalStatus build();
    BuildStep id(String id);
    BuildStep goalText(String goalText);
    BuildStep recordedDate(Temporal.DateTime recordedDate);
    BuildStep goalAdditionalInfo(String goalAdditionalInfo);
  }
  

  public static class Builder implements UsernameStep, VersionStep, GoalReferenceStep, TaskReferencesStep, StartDateStep, DueDateStep, CompleteStep, SelectedStep, BuildStep {
    private String id;
    private String username;
    private Integer version;
    private String goalReference;
    private List<String> taskReferences;
    private Temporal.DateTime startDate;
    private Temporal.DateTime dueDate;
    private Boolean complete;
    private Boolean selected;
    private String goalText;
    private Temporal.DateTime recordedDate;
    private String goalAdditionalInfo;
    @Override
     public GoalStatus build() {
        String id = this.id != null ? this.id : UUID.randomUUID().toString();
        
        return new GoalStatus(
          id,
          username,
          version,
          goalReference,
          goalText,
          taskReferences,
          startDate,
          dueDate,
          recordedDate,
          complete,
          selected,
          goalAdditionalInfo);
    }
    
    @Override
     public VersionStep username(String username) {
        Objects.requireNonNull(username);
        this.username = username;
        return this;
    }
    
    @Override
     public GoalReferenceStep version(Integer version) {
        Objects.requireNonNull(version);
        this.version = version;
        return this;
    }
    
    @Override
     public TaskReferencesStep goalReference(String goalReference) {
        Objects.requireNonNull(goalReference);
        this.goalReference = goalReference;
        return this;
    }
    
    @Override
     public StartDateStep taskReferences(List<String> taskReferences) {
        Objects.requireNonNull(taskReferences);
        this.taskReferences = taskReferences;
        return this;
    }
    
    @Override
     public DueDateStep startDate(Temporal.DateTime startDate) {
        Objects.requireNonNull(startDate);
        this.startDate = startDate;
        return this;
    }
    
    @Override
     public CompleteStep dueDate(Temporal.DateTime dueDate) {
        Objects.requireNonNull(dueDate);
        this.dueDate = dueDate;
        return this;
    }
    
    @Override
     public SelectedStep complete(Boolean complete) {
        Objects.requireNonNull(complete);
        this.complete = complete;
        return this;
    }
    
    @Override
     public BuildStep selected(Boolean selected) {
        Objects.requireNonNull(selected);
        this.selected = selected;
        return this;
    }
    
    @Override
     public BuildStep goalText(String goalText) {
        this.goalText = goalText;
        return this;
    }
    
    @Override
     public BuildStep recordedDate(Temporal.DateTime recordedDate) {
        this.recordedDate = recordedDate;
        return this;
    }
    
    @Override
     public BuildStep goalAdditionalInfo(String goalAdditionalInfo) {
        this.goalAdditionalInfo = goalAdditionalInfo;
        return this;
    }
    
    /** 
     * @param id id
     * @return Current Builder instance, for fluent method chaining
     */
    public BuildStep id(String id) {
        this.id = id;
        return this;
    }
  }
  

  public final class CopyOfBuilder extends Builder {
    private CopyOfBuilder(String id, String username, Integer version, String goalReference, String goalText, List<String> taskReferences, Temporal.DateTime startDate, Temporal.DateTime dueDate, Temporal.DateTime recordedDate, Boolean complete, Boolean selected, String goalAdditionalInfo) {
      super.id(id);
      super.username(username)
        .version(version)
        .goalReference(goalReference)
        .startDate(startDate)
        .dueDate(dueDate)
        .complete(complete)
        .selected(selected)
        .goalText(goalText)
        .taskReferences(taskReferences)
        .recordedDate(recordedDate)
        .goalAdditionalInfo(goalAdditionalInfo);
    }
    
    @Override
     public CopyOfBuilder username(String username) {
      return (CopyOfBuilder) super.username(username);
    }
    
    @Override
     public CopyOfBuilder version(Integer version) {
      return (CopyOfBuilder) super.version(version);
    }
    
    @Override
     public CopyOfBuilder goalReference(String goalReference) {
      return (CopyOfBuilder) super.goalReference(goalReference);
    }
    
    @Override
     public CopyOfBuilder startDate(Temporal.DateTime startDate) {
      return (CopyOfBuilder) super.startDate(startDate);
    }
    
    @Override
     public CopyOfBuilder dueDate(Temporal.DateTime dueDate) {
      return (CopyOfBuilder) super.dueDate(dueDate);
    }
    
    @Override
     public CopyOfBuilder complete(Boolean complete) {
      return (CopyOfBuilder) super.complete(complete);
    }
    
    @Override
     public CopyOfBuilder selected(Boolean selected) {
      return (CopyOfBuilder) super.selected(selected);
    }
    
    @Override
     public CopyOfBuilder goalText(String goalText) {
      return (CopyOfBuilder) super.goalText(goalText);
    }
    
    @Override
     public CopyOfBuilder taskReferences(List<String> taskReferences) {
      return (CopyOfBuilder) super.taskReferences(taskReferences);
    }
    
    @Override
     public CopyOfBuilder recordedDate(Temporal.DateTime recordedDate) {
      return (CopyOfBuilder) super.recordedDate(recordedDate);
    }
    
    @Override
     public CopyOfBuilder goalAdditionalInfo(String goalAdditionalInfo) {
      return (CopyOfBuilder) super.goalAdditionalInfo(goalAdditionalInfo);
    }
  }
  
}

```



### Reproduction steps (if applicable)

I have created one schema with the name of GoalStatus. But when I run the command **amplify codegen models**
Model class generated but it gives an error.

```
/app/src/main/java/com/amplifyframework/datastore/generated/model/GoalStatus.java:408: error: cannot find symbol
        .startDate(startDate)
        ^
  symbol:   method startDate(DateTime)
  location: interface TaskReferencesStep


```
![Screenshot from 2022-04-13 11-44-10](https://user-images.githubusercontent.com/13220264/163112389-0507eb78-1544-4cc1-abb0-83bc216c3941.png)



```
type GoalStatus @model @auth(rules: [{ allow: owner }]) {
id: ID!
username: String!
version: Int!
goalReference: String!
goalText: String
taskReferences: [String]!
startDate: AWSDateTime!
dueDate: AWSDateTime!
recordedDate: AWSDateTime
complete: Boolean!
selected: Boolean!
goalAdditionalInfo:String
}
```
```
package com.amplifyframework.datastore.generated.model;

import com.amplifyframework.core.model.temporal.Temporal;

import java.util.List;
import java.util.UUID;
import java.util.Objects;

import androidx.core.util.ObjectsCompat;

import com.amplifyframework.core.model.AuthStrategy;
import com.amplifyframework.core.model.Model;
import com.amplifyframework.core.model.ModelOperation;
import com.amplifyframework.core.model.annotations.AuthRule;
import com.amplifyframework.core.model.annotations.Index;
import com.amplifyframework.core.model.annotations.ModelConfig;
import com.amplifyframework.core.model.annotations.ModelField;
import com.amplifyframework.core.model.query.predicate.QueryField;

import static com.amplifyframework.core.model.query.predicate.QueryField.field;

/** This is an auto generated class representing the GoalStatus type in your schema. */
@SuppressWarnings("all")
@ModelConfig(pluralName = "GoalStatuses", authRules = {
  @AuthRule(allow = AuthStrategy.OWNER, ownerField = "owner", identityClaim = "cognito:username", provider = "userPools", operations = { ModelOperation.CREATE, ModelOperation.UPDATE, ModelOperation.DELETE, ModelOperation.READ })
})
public final class GoalStatus implements Model {
  public static final QueryField ID = field("GoalStatus", "id");
  public static final QueryField USERNAME = field("GoalStatus", "username");
  public static final QueryField VERSION = field("GoalStatus", "version");
  public static final QueryField GOAL_REFERENCE = field("GoalStatus", "goalReference");
  public static final QueryField GOAL_TEXT = field("GoalStatus", "goalText");
  public static final QueryField TASK_REFERENCES = field("GoalStatus", "taskReferences");
  public static final QueryField START_DATE = field("GoalStatus", "startDate");
  public static final QueryField DUE_DATE = field("GoalStatus", "dueDate");
  public static final QueryField RECORDED_DATE = field("GoalStatus", "recordedDate");
  public static final QueryField COMPLETE = field("GoalStatus", "complete");
  public static final QueryField SELECTED = field("GoalStatus", "selected");
  public static final QueryField GOAL_ADDITIONAL_INFO = field("GoalStatus", "goalAdditionalInfo");
  private final @ModelField(targetType="ID", isRequired = true) String id;
  private final @ModelField(targetType="String", isRequired = true) String username;
  private final @ModelField(targetType="Int", isRequired = true) Integer version;
  private final @ModelField(targetType="String", isRequired = true) String goalReference;
  private final @ModelField(targetType="String") String goalText;
  private final @ModelField(targetType="String", isRequired = true) List<String> taskReferences;
  private final @ModelField(targetType="AWSDateTime", isRequired = true) Temporal.DateTime startDate;
  private final @ModelField(targetType="AWSDateTime", isRequired = true) Temporal.DateTime dueDate;
  private final @ModelField(targetType="AWSDateTime") Temporal.DateTime recordedDate;
  private final @ModelField(targetType="Boolean", isRequired = true) Boolean complete;
  private final @ModelField(targetType="Boolean", isRequired = true) Boolean selected;
  private final @ModelField(targetType="String") String goalAdditionalInfo;
  private @ModelField(targetType="AWSDateTime", isReadOnly = true) Temporal.DateTime createdAt;
  private @ModelField(targetType="AWSDateTime", isReadOnly = true) Temporal.DateTime updatedAt;
  public String getId() {
      return id;
  }
  
  public String getUsername() {
      return username;
  }
  
  public Integer getVersion() {
      return version;
  }
  
  public String getGoalReference() {
      return goalReference;
  }
  
  public String getGoalText() {
      return goalText;
  }
  
  public List<String> getTaskReferences() {
      return taskReferences;
  }
  
  public Temporal.DateTime getStartDate() {
      return startDate;
  }
  
  public Temporal.DateTime getDueDate() {
      return dueDate;
  }
  
  public Temporal.DateTime getRecordedDate() {
      return recordedDate;
  }
  
  public Boolean getComplete() {
      return complete;
  }
  
  public Boolean getSelected() {
      return selected;
  }
  
  public String getGoalAdditionalInfo() {
      return goalAdditionalInfo;
  }
  
  public Temporal.DateTime getCreatedAt() {
      return createdAt;
  }
  
  public Temporal.DateTime getUpdatedAt() {
      return updatedAt;
  }
  
  private GoalStatus(String id, String username, Integer version, String goalReference, String goalText, List<String> taskReferences, Temporal.DateTime startDate, Temporal.DateTime dueDate, Temporal.DateTime recordedDate, Boolean complete, Boolean selected, String goalAdditionalInfo) {
    this.id = id;
    this.username = username;
    this.version = version;
    this.goalReference = goalReference;
    this.goalText = goalText;
    this.taskReferences = taskReferences;
    this.startDate = startDate;
    this.dueDate = dueDate;
    this.recordedDate = recordedDate;
    this.complete = complete;
    this.selected = selected;
    this.goalAdditionalInfo = goalAdditionalInfo;
  }
  
  @Override
   public boolean equals(Object obj) {
      if (this == obj) {
        return true;
      } else if(obj == null || getClass() != obj.getClass()) {
        return false;
      } else {
      GoalStatus goalStatus = (GoalStatus) obj;
      return ObjectsCompat.equals(getId(), goalStatus.getId()) &&
              ObjectsCompat.equals(getUsername(), goalStatus.getUsername()) &&
              ObjectsCompat.equals(getVersion(), goalStatus.getVersion()) &&
              ObjectsCompat.equals(getGoalReference(), goalStatus.getGoalReference()) &&
              ObjectsCompat.equals(getGoalText(), goalStatus.getGoalText()) &&
              ObjectsCompat.equals(getTaskReferences(), goalStatus.getTaskReferences()) &&
              ObjectsCompat.equals(getStartDate(), goalStatus.getStartDate()) &&
              ObjectsCompat.equals(getDueDate(), goalStatus.getDueDate()) &&
              ObjectsCompat.equals(getRecordedDate(), goalStatus.getRecordedDate()) &&
              ObjectsCompat.equals(getComplete(), goalStatus.getComplete()) &&
              ObjectsCompat.equals(getSelected(), goalStatus.getSelected()) &&
              ObjectsCompat.equals(getGoalAdditionalInfo(), goalStatus.getGoalAdditionalInfo()) &&
              ObjectsCompat.equals(getCreatedAt(), goalStatus.getCreatedAt()) &&
              ObjectsCompat.equals(getUpdatedAt(), goalStatus.getUpdatedAt());
      }
  }
  
  @Override
   public int hashCode() {
    return new StringBuilder()
      .append(getId())
      .append(getUsername())
      .append(getVersion())
      .append(getGoalReference())
      .append(getGoalText())
      .append(getTaskReferences())
      .append(getStartDate())
      .append(getDueDate())
      .append(getRecordedDate())
      .append(getComplete())
      .append(getSelected())
      .append(getGoalAdditionalInfo())
      .append(getCreatedAt())
      .append(getUpdatedAt())
      .toString()
      .hashCode();
  }
  
  @Override
   public String toString() {
    return new StringBuilder()
      .append("GoalStatus {")
      .append("id=" + String.valueOf(getId()) + ", ")
      .append("username=" + String.valueOf(getUsername()) + ", ")
      .append("version=" + String.valueOf(getVersion()) + ", ")
      .append("goalReference=" + String.valueOf(getGoalReference()) + ", ")
      .append("goalText=" + String.valueOf(getGoalText()) + ", ")
      .append("taskReferences=" + String.valueOf(getTaskReferences()) + ", ")
      .append("startDate=" + String.valueOf(getStartDate()) + ", ")
      .append("dueDate=" + String.valueOf(getDueDate()) + ", ")
      .append("recordedDate=" + String.valueOf(getRecordedDate()) + ", ")
      .append("complete=" + String.valueOf(getComplete()) + ", ")
      .append("selected=" + String.valueOf(getSelected()) + ", ")
      .append("goalAdditionalInfo=" + String.valueOf(getGoalAdditionalInfo()) + ", ")
      .append("createdAt=" + String.valueOf(getCreatedAt()) + ", ")
      .append("updatedAt=" + String.valueOf(getUpdatedAt()))
      .append("}")
      .toString();
  }
  
  public static UsernameStep builder() {
      return new Builder();
  }
  
  /** 
   * WARNING: This method should not be used to build an instance of this object for a CREATE mutation.
   * This is a convenience method to return an instance of the object with only its ID populated
   * to be used in the context of a parameter in a delete mutation or referencing a foreign key
   * in a relationship.
   * @param id the id of the existing item this instance will represent
   * @return an instance of this model with only ID populated
   */
  public static GoalStatus justId(String id) {
    return new GoalStatus(
      id,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null
    );
  }
  
  public CopyOfBuilder copyOfBuilder() {
    return new CopyOfBuilder(id,
      username,
      version,
      goalReference,
      goalText,
      taskReferences,
      startDate,
      dueDate,
      recordedDate,
      complete,
      selected,
      goalAdditionalInfo);
  }
  public interface UsernameStep {
    VersionStep username(String username);
  }
  

  public interface VersionStep {
    GoalReferenceStep version(Integer version);
  }
  

  public interface GoalReferenceStep {
    TaskReferencesStep goalReference(String goalReference);
  }
  

  public interface TaskReferencesStep {
    StartDateStep taskReferences(List<String> taskReferences);
  }
  

  public interface StartDateStep {
    DueDateStep startDate(Temporal.DateTime startDate);
  }
  

  public interface DueDateStep {
    CompleteStep dueDate(Temporal.DateTime dueDate);
  }
  

  public interface CompleteStep {
    SelectedStep complete(Boolean complete);
  }
  

  public interface SelectedStep {
    BuildStep selected(Boolean selected);
  }
  

  public interface BuildStep {
    GoalStatus build();
    BuildStep id(String id);
    BuildStep goalText(String goalText);
    BuildStep recordedDate(Temporal.DateTime recordedDate);
    BuildStep goalAdditionalInfo(String goalAdditionalInfo);
  }
  

  public static class Builder implements UsernameStep, VersionStep, GoalReferenceStep, TaskReferencesStep, StartDateStep, DueDateStep, CompleteStep, SelectedStep, BuildStep {
    private String id;
    private String username;
    private Integer version;
    private String goalReference;
    private List<String> taskReferences;
    private Temporal.DateTime startDate;
    private Temporal.DateTime dueDate;
    private Boolean complete;
    private Boolean selected;
    private String goalText;
    private Temporal.DateTime recordedDate;
    private String goalAdditionalInfo;
    @Override
     public GoalStatus build() {
        String id = this.id != null ? this.id : UUID.randomUUID().toString();
        
        return new GoalStatus(
          id,
          username,
          version,
          goalReference,
          goalText,
          taskReferences,
          startDate,
          dueDate,
          recordedDate,
          complete,
          selected,
          goalAdditionalInfo);
    }
    
    @Override
     public VersionStep username(String username) {
        Objects.requireNonNull(username);
        this.username = username;
        return this;
    }
    
    @Override
     public GoalReferenceStep version(Integer version) {
        Objects.requireNonNull(version);
        this.version = version;
        return this;
    }
    
    @Override
     public TaskReferencesStep goalReference(String goalReference) {
        Objects.requireNonNull(goalReference);
        this.goalReference = goalReference;
        return this;
    }
    
    @Override
     public StartDateStep taskReferences(List<String> taskReferences) {
        Objects.requireNonNull(taskReferences);
        this.taskReferences = taskReferences;
        return this;
    }
    
    @Override
     public DueDateStep startDate(Temporal.DateTime startDate) {
        Objects.requireNonNull(startDate);
        this.startDate = startDate;
        return this;
    }
    
    @Override
     public CompleteStep dueDate(Temporal.DateTime dueDate) {
        Objects.requireNonNull(dueDate);
        this.dueDate = dueDate;
        return this;
    }
    
    @Override
     public SelectedStep complete(Boolean complete) {
        Objects.requireNonNull(complete);
        this.complete = complete;
        return this;
    }
    
    @Override
     public BuildStep selected(Boolean selected) {
        Objects.requireNonNull(selected);
        this.selected = selected;
        return this;
    }
    
    @Override
     public BuildStep goalText(String goalText) {
        this.goalText = goalText;
        return this;
    }
    
    @Override
     public BuildStep recordedDate(Temporal.DateTime recordedDate) {
        this.recordedDate = recordedDate;
        return this;
    }
    
    @Override
     public BuildStep goalAdditionalInfo(String goalAdditionalInfo) {
        this.goalAdditionalInfo = goalAdditionalInfo;
        return this;
    }
    
    /** 
     * @param id id
     * @return Current Builder instance, for fluent method chaining
     */
    public BuildStep id(String id) {
        this.id = id;
        return this;
    }
  }
  

  public final class CopyOfBuilder extends Builder {
    private CopyOfBuilder(String id, String username, Integer version, String goalReference, String goalText, List<String> taskReferences, Temporal.DateTime startDate, Temporal.DateTime dueDate, Temporal.DateTime recordedDate, Boolean complete, Boolean selected, String goalAdditionalInfo) {
      super.id(id);
      super.username(username)
        .version(version)
        .goalReference(goalReference)
        .startDate(startDate)
        .dueDate(dueDate)
        .complete(complete)
        .selected(selected)
        .goalText(goalText)
        .taskReferences(taskReferences)
        .recordedDate(recordedDate)
        .goalAdditionalInfo(goalAdditionalInfo);
    }
    
    @Override
     public CopyOfBuilder username(String username) {
      return (CopyOfBuilder) super.username(username);
    }
    
    @Override
     public CopyOfBuilder version(Integer version) {
      return (CopyOfBuilder) super.version(version);
    }
    
    @Override
     public CopyOfBuilder goalReference(String goalReference) {
      return (CopyOfBuilder) super.goalReference(goalReference);
    }
    
    @Override
     public CopyOfBuilder startDate(Temporal.DateTime startDate) {
      return (CopyOfBuilder) super.startDate(startDate);
    }
    
    @Override
     public CopyOfBuilder dueDate(Temporal.DateTime dueDate) {
      return (CopyOfBuilder) super.dueDate(dueDate);
    }
    
    @Override
     public CopyOfBuilder complete(Boolean complete) {
      return (CopyOfBuilder) super.complete(complete);
    }
    
    @Override
     public CopyOfBuilder selected(Boolean selected) {
      return (CopyOfBuilder) super.selected(selected);
    }
    
    @Override
     public CopyOfBuilder goalText(String goalText) {
      return (CopyOfBuilder) super.goalText(goalText);
    }
    
    @Override
     public CopyOfBuilder taskReferences(List<String> taskReferences) {
      return (CopyOfBuilder) super.taskReferences(taskReferences);
    }
    
    @Override
     public CopyOfBuilder recordedDate(Temporal.DateTime recordedDate) {
      return (CopyOfBuilder) super.recordedDate(recordedDate);
    }
    
    @Override
     public CopyOfBuilder goalAdditionalInfo(String goalAdditionalInfo) {
      return (CopyOfBuilder) super.goalAdditionalInfo(goalAdditionalInfo);
    }
  }
  
}

```



### Code Snippet

```java
// Put your code below this line.

```


### Log output

<details>

```
// Put your logs below this line


```

</details>


### amplifyconfiguration.json

_No response_

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>


### Additional information and screenshots

_No response_

### Expected behavior

https://github.com/aws-amplify/amplify-android/issues/1702

### Reproduction steps

https://github.com/aws-amplify/amplify-android/issues/1702

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>


### Log output

<details>

```
# Put your logs below this line


```

</details>


### Additional information

_No response_

--- a/packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts
+++ b/packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts
@@ -513,9 +513,9 @@ export class AppSyncModelJavaVisitor<
       .extends(['Builder']);
 
     const nonNullableFields = this.getWritableFields(model)
-      .filter(field => !field.isNullable)
+      .filter(field => this.isRequiredField(field))
       .filter(f => (isModel ? f.name !== 'id' : true));
-    const nullableFields = this.getWritableFields(model).filter(field => field.isNullable);
+    const nullableFields = this.getWritableFields(model).filter(field => !this.isRequiredField(field));
 
     // constructor
     const constructorArguments = this.getWritableFields(model).map(field => ({

